### PR TITLE
refactor: handle toastr notifications in middleware

### PR DESCRIPTION
**Pull request by pyphilia**
This PR aims to refactor toastr notifications (compared to graasp-desktop) in a common middleware.

In doing so, `types` had to be accessible from the `public` folder. I therefore took this opportunity to handle duplicates of `channels.js`, `errors.js`, `messages.js`, etc. With this PR, these files are located in `src/shared` and are duplicated in `public/shared` before each `yarn start` and `yarn mocha` (inspired by https://medium.com/@johndyer24/building-a-production-electron-create-react-app-application-with-shared-code-using-electron-builder-c1f70f0e2649). 

The main changes implied:
1. Actions do not create the action objects, listeners do. Success actions are unchanged. Error actions have the following shape 
```
{
  type: GET_DATASET_ERROR, 
  error: ERROR_GENERAL,
  payload?: {somedata}
}
```
where payload can be used to customize the message error or change data in the reducer.

2. `channels.js`, `errors.js`, `messages.js` and such are only edited in `src/shared` (changes are reported automatically). `types.js` is a single file instead of a folder.
3. `toastrMiddleware.js` receives all dispatched actions, one just needs to catch them in the switch when a notification is required.

With the current changes, we end up with relatively repetitive and simple actions. We might create a utility function to generate them if necessary. 

Duplicating files is not the ideal solution. Another solution I looked into was to use the public folder in a react way https://create-react-app.dev/docs/using-the-public-folder/ but it does not seem to support dynamic import (such as `import x from 'PUBLIC_URL+filename'`).

Let me know about what you think about this solution.

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -82,15 +84,24 @@
...
+    "dist:linux": "env-cmd -f ./.env electron-builder -l",
+    "dist:posix": "env-cmd -f ./.env electron-builder -ml",
+    "dist:win:docker": "./scripts/buildWindowsWithDocker.sh",
+    "dist:all": "run-s dist:posix dist:win",
     "lint": "eslint .",
     "prestart": "env-cmd -f ./.env.local node scripts/setup.js",
-    "start": "concurrently \"env-cmd -f ./.env.local react-scripts start\" \"wait-on http://localhost:3000 && env-cmd -f ./.env.local electron .\"",
-    "mocha": "concurrently \"env-cmd -f ./.env.test react-scripts start\" \"wait-on http://localhost:3000 && mocha --require @babel/register \"test/**/*.test.js\"\"",
+    "start": "concurrently \"yarn:prepare\" \"env-cmd -f ./.env.local react-scripts start\" \"wait-on http://localhost:3000 && env-cmd -f ./.env.local electron .\"",
+    "mocha": "concurrently \"yarn:prepare\" \"env-cmd -f ./.env.test react-scripts start\" \"wait-on http://localhost:3000 && mocha --require @babel/register \"test/**/*.test.js\"\"",
```
[the remark this answers is not in the document]
Comment (juancarlosfarah) [reply package.json:98]: Don't think you need the `:` between `yarn` and `prepare`.

**Review by juancarlosfarah — CHANGES_REQUESTED** [review]
Very nice! Just some minor comments and questions.

**Comment by pyphilia** [comment]
I applied the changes and rebased on the master to include all the changes.

**Comment by pyphilia** [comment]
I've rebased again, it would be nice to merge this one!

**Review by hagoptaminian — APPROVED** [review]
@pyphilia this looks good to me. nice work.

**Pull request merged**